### PR TITLE
Complete Phase 3.2: Contract Tests (T006-T012) - TDD

### PR DESCRIPTION
**Pull request by ddumdi11**
Implemented all API contract tests following Test-Driven Development:

**T006-T008: Interrogation API Tests**
- POST /interrogation/start - 11 test cases
  - Request validation (hypothesis, iterationLimit, detectiveProvider, witnessModel)
  - Session state validation (FR-029 - single active session)
  - Response schema validation (UUID v4 sessionId, status)
- POST /interrogation/stop - 9 test cases
  - Session state validation (404 for non-existent/stopped sessions)
  - Partial results return (FR-020)
- GET /interrogation/status/:sessionId - 13 test cases
  - Real-time progress updates (FR-011, FR-012)
  - Status transitions (running → completed/failed/limit-reached)

**T009-T010: Session Management Tests**
- GET /sessions - 6 test cases
  - Session list with metadata (FR-022)
  - Ordering by startTime descending
  - ConsistencyScore only for completed sessions
- GET /sessions/:sessionId - 8 test cases
  - Complete InterrogationSession schema (FR-023)
  - Data integrity validation
  - Audit trail preservation

**T011-T012: Configuration API Tests**
- PUT /config/credentials - 12 test cases
  - Multi-provider support (OpenAI, Anthropic, Gemini)
  - API key encryption (FR-030)
  - Credential validation (FR-005)
  - Default provider management
- PUT /config/timeouts - 13 test cases
  - Cloud timeout validation (15-120s, FR-034)
  - Local timeout validation (15-360s, FR-035)
  - Persistence across sessions

**Test Infrastructure:**
- Created tests/setup.ts with IPC mock (global.ipcRenderer)
- Updated jest.config.main.js with setupFilesAfterEnv
- All 70+ tests FAIL as expected (TDD ✓)

**Files Created:**
- tests/setup.ts
- tests/unit/api/interrogation-start.test.ts
- tests/unit/api/interrogation-stop.test.ts
- tests/unit/api/interrogation-status.test.ts
- tests/unit/api/session-list.test.ts
- tests/unit/api/session-load.test.ts
- tests/unit/api/config-credentials.test.ts
- tests/unit/api/config-timeouts.test.ts

**Verified:**
✓ All tests fail with "IPC handler not implemented" (expected in TDD) ✓ Test structure matches OpenAPI contracts
✓ Coverage requirements aligned (60% threshold for agents/api)

Ready for Phase 3.3 (Implementation) - tests will guide development.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

- Tests
  - Added comprehensive unit tests for configuration (credentials, timeouts), interrogation flows (start, status, stop), and sessions (list, load), covering validation, persistence, ordering, response schemas, real-time progress, partial results, and edge cases.

- Chores
  - Centralized test environment setup with a global IPC mock loaded after test init.
  - Added UUID-related dependencies for testing and validation.

- Note: No user-facing changes in this release.

### CODE REVIEW

**Review by ddumdi11 — COMMENTED** [review]
Gute Arbeit, sauber und es geht gut und schnell voran. Vielen Dank.

File: tests/unit/api/session-list.test.ts [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+      const listResponse = await global.ipcRenderer.invoke('sessions:list');
+
+      // Verify ordering (newest first)
+      for (let i = 1; i < listResponse.sessions.length; i++) {
+        const prev = new Date(listResponse.sessions[i - 1].startTime);
+        const curr = new Date(listResponse.sessions[i].startTime);
+        expect(prev >= curr).toBe(true);
+      }
+    });
+  });
```
[the remark this answers is not in the document]
Comment (ddumdi11) [reply tests/unit/api/session-list.test.ts:120]: Korrigiert!

File: tests/unit/api/session-load.test.ts [hunk 2]
```diff
@@ -0,0 +1,258 @@
...
+            endTime: new Date().toISOString(),
+            status: 'failed',
+            iterationLimit: 5,
+            currentIteration: mockQAPairs.length,
+            qaPairs: mockQAPairs,
+            auditTrail: [],
+          });
+        }
+        return Promise.reject(new Error(`Unhandled channel: ${channel}`));
+      });
```
[the remark this answers is not in the document]
Comment (ddumdi11) [reply tests/unit/api/session-load.test.ts:171]: Erledigt.

File: tests/unit/api/interrogation-stop.test.ts [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+        witnessModel: 'llama2',
+      });
+
+      const stopResponse = await global.ipcRenderer.invoke('interrogation:stop', {
+        sessionId: startResponse.sessionId,
+      });
+
+      expect(stopResponse.partialResults.qaPairs).toBeInstanceOf(Array);
+      expect(stopResponse.partialResults.currentIteration).toBeGreaterThanOrEqual(0);
+    });
```
[the remark this answers is not in the document]
Comment (ddumdi11) [reply tests/unit/api/interrogation-stop.test.ts:176]: Erledigt.
